contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/schema/HiveDatabaseSchema.java

```diff
@@ -72,4 +80,76 @@ public String getTypeName() {
     return HiveStoragePluginConfig.NAME;
   }
 
+  @Override
+  public List<Pair<String, ? extends Table>> getTablesByNames(final List<String> tableNames) {
+    final String schemaName = getName();
+    final List<Pair<String, ? extends Table>> tableNameToTable = Lists.newArrayList();
+    List<org.apache.hadoop.hive.metastore.api.Table> tables;
+    // Retries once if the first call to fetch the metadata fails
+    synchronized(mClient) {
+      final List<String> tableNamesWithAuth = Lists.newArrayList();
+      for(String tableName : tableNames) {
+        try {
+          if(mClient.tableExists(schemaName, tableName)) {
```

Comment:
I think we should fix the issue in SQL based authorization to show only tables that the user has access to. Don't recall why it showed all tables. We can use the same Hive authorizer used to check the permission when querying on Hive table.